### DISCUSSION TITLE
Bulk import of CSV or JSON  file fails with 413 Payload Too Large error

### DISCUSSION THREAD

**Discussion by KE8SOJ** [sys4/netbox-plugin-dns#272] [Q&A] [answered]
I receive a 413 "Payload To Large" error when attempting a bulk import of a file (CSV or JSON format) DNS records into NetBox. While this may be related to the server's http proxy settings, does the DNS plugin have a default maximum payload size value? If so can the the maximum payload size value be increased in the Django configuration file?

```
$ ls -hl dc6-prod_records_dns.csv
55M May 23 12:19 dc6-prod_records_dns.csv

$ wc -l dc6-prod_records_dns.csv
727603 dc6-prod_records_dns.csv
```

**Comment by peteeckel** [comment]
Hi @KE8SOJ, that's not a NetBox DNS issue (nor NetBox, by the way), not even Django - it's your web server/reverse proxy config.

Are you using Apache or Nginx?

**Comment by peteeckel** [comment]
You can find instructitions for properly configuring Apache or Nginx [here](https://computingforgeeks.com/how-to-resolve-413-request-entity-too-large-error-on-nginx/) (and in about a million other places, if you know what to look for :-))

**Comment by peteeckel** [comment, marked as answer]
Update: There actually **is** a limit in Django, with a config setting in NetBox, which is 2.5 MB (2621440 bytes) by default. It was introduced in NetBox version 3.6. See the [Django documentation](https://docs.djangoproject.com/en/5.0/ref/settings/#data-upload-max-memory-size)

You can increase `DATA_UPLOAD_MAX_MEMORY_SIZE` in `configuration.py` and restart the `netbox` and `netbox-rq` services. Setting it to `None` disables the check.

**Reply by KE8SOJ** [reply]
Thank you. I succeeded in importing the CSV file after setting the  `DATA_UPLOAD_MAX_MEMORY_SIZE = 104857600` and configuring the http-proxy (NGINX-unit). Hope this helps.
```
  "settings": {
    "http": {
      "max_body_size": 104857600,
      "body_read_timeout": 120
    }
  },
```

**Comment by peteeckel** [comment]
Moved this to a discussion to keep it in focus and because it's not an issue after all.
